maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/RawXJC2Mojo.java

```diff
@@ -305,26 +306,26 @@ public List<URI> getProducesURIs() {
 	 */
 	public void execute() throws MojoExecutionException {
 		synchronized (lock) {
-			injectDependencyDefaults();
-			resolveArtifacts();
-
-			// Install project dependencies into classloader's class path
-			// and execute xjc2.
-			final ClassLoader currentClassLoader = Thread.currentThread().getContextClassLoader();
-			final ClassLoader classLoader = createClassLoader(currentClassLoader);
-			Thread.currentThread().setContextClassLoader(classLoader);
-			final Locale currentDefaultLocale = Locale.getDefault();
-			try {
-				final Locale locale = LocaleUtils.valueOf(getLocale());
-				Locale.setDefault(locale);
-				//
-				doExecute();
-
-			} finally {
-				Locale.setDefault(currentDefaultLocale);
-				// Set back the old classloader
-				Thread.currentThread().setContextClassLoader(currentClassLoader);
-			}
+            injectDependencyDefaults();
+            resolveArtifacts();
+
+            // Install project dependencies into classloader's class path and execute xjc2.
+            final ClassLoader currentClassLoader = Thread.currentThread().getContextClassLoader();
+            final Locale currentDefaultLocale = Locale.getDefault();
+            URLClassLoader classLoader = null;
```

Comment:
URLClassLoader implements Closeable. We should use try-with-resource instead.

try (URLClassLoader classLoader .. ) {

Reply:
Using try with resource pattern let me have another new IOException to handle on the close part.
Didnt know what to do with it do I set it to utils to try close with ignoring exception like in IOUtils plexus method

Reply:
Should looks like what expected now 😄